controller/src/main/java/io/pravega/controller/store/PravegaTablesScope.java

```diff
@@ -180,37 +217,77 @@ public String getName() {
+                                    return CompletableFuture.allOf(storeHelper.deleteTable(streamsInScopeTableName, true, context.getRequestId()),
+                                                                   storeHelper.deleteTable(kvtsInScopeTableName, true, context.getRequestId()),
+                                                                   storeHelper.deleteTable(rgsInScopeTableName, true, context.getRequestId()),
+                                                                   Futures.allOf(deleteTagTablesFut))
+                                                            .thenAccept(v -> log.debug("tables deleted {} {} {}", streamsInScopeTableName,
+                                                                                       kvtsInScopeTableName, rgsInScopeTableName));
+                                })
+                                .thenCompose(deleted -> storeHelper.removeEntry(SCOPES_TABLE, scopeName, context.getRequestId()));
     }
 
     @Override
-    public CompletableFuture<Pair<List<String>, String>> listStreams(int limit, String continuationToken, Executor executor, 
+    public CompletableFuture<Pair<List<String>, String>> listStreams(int limit, String continuationToken, Executor executor,
                                                                      OperationContext context) {
         Preconditions.checkNotNull(context, "Operation context cannot be null");
         return getStreamsInScopeTableName(context)
                 .thenCompose(streamsInScopeTable -> readAll(limit, continuationToken, streamsInScopeTable, context));
     }
 
+    @Override
+    public CompletableFuture<Pair<List<String>, String>> listStreamsForTag(String tag, String continuationToken, Executor executor,
+                                                                           OperationContext context) {
+
+        Preconditions.checkNotNull(context, "Operation context cannot be null");
+
+        return getStreamsFromNextTagChunk(tag, continuationToken, context).thenCompose(pair -> {
+            if (pair.getLeft().isEmpty() && !pair.getRight().contains(LAST_TAG_CHUNK)) {
+                return listStreamsForTag(tag, pair.getRight(), executor, context);
+            } else {
+                return CompletableFuture.completedFuture(pair);
+            }
+        });
+    }
+
+    /*
+        Fetch the Streams from the next available Tag Chunk table.
+     */
+    CompletableFuture<Pair<List<String>, String>> getStreamsFromNextTagChunk(String tag, String token, OperationContext context) {
+        return getAllStreamTagsInScopeTableNames(context).thenApply(
+                strings -> {
+                    if (token.isEmpty()) {
+                        // token is empty, try reading from the first tag table.
+                        return strings.get(0);
+                    } else {
+                        // return next index
+                        return strings.get(strings.indexOf(token) + 1);
```

Comment:
how are we using the continuation token here?
strings refers to all the table chunk names, right?
token is a particular chunk number, right?
why do we need to do strings.indexOf(token)?

Reply:
updated the java docs.